File: services/main-service/Domain/Entities/Issue.cs
namespace MainService.Domain.Entities;

public class IssueDomain
{
    public string? Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string ProjectId { get; set; } = string.Empty;
    public string? SprintId { get; set; }
    public string? AssigneeId { get; set; }
    public string ParentId { get; set; } = string.Empty;
    public string ReporterId { get; private set; } = string.Empty;
    public IssueType Type { get; set; }
    public IssueStatus Status { get; set; }
    public IssuePriority Priority { get; set; } = IssuePriority.Medium;

    public string Summary { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int StoryPoint { get; set; }
    public List<string> Attachments { get; set; } = new();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public IssueDomain(string projectId, string reporterId)
    {
        ProjectId = projectId;
        ReporterId = reporterId;
    }

    public void AssignToSprint(string? sprintId)
    {
        SprintId = sprintId;
        UpdatedAt = DateTime.UtcNow;
    }

    public void AssignToUser(string? assigneeId)
    {
        AssigneeId = assigneeId;
        UpdatedAt = DateTime.UtcNow;
    }
}

public enum IssueType
{
    Bug,
    Task,
    Story,
    Epic
}

public enum IssueStatus
{
    ToDo,
    InProgress,
    Done,
    Closed
}

public enum IssuePriority
{
    Low,
    Medium,
    High,
    Critical
}

File: services/express-service/src/models/issue.js
import mongoose from 'mongoose'

const IssueSchema = new mongoose.Schema(
  {
    _id: { type: String }, // nếu bạn tự generate, không cần ObjectId
    title: { type: String, default: '' },
    project_id: { type: String, required: true },
    sprint_id: { type: String, default: '' },
    assignee_id: { type: String, default: '' },
    parent_id: { type: String, default: '' },
    reporter_id: { type: String, required: true },
    type: { type: String, default: 'Task' }, // Epic | Story | Task | Bug
    column_id: { type: String, default: '' },
    column: { type: String, default: null },
    priority: { type: String, default: 'Medium' },
    summary: { type: String, default: '' },
    description: { type: String, default: '' }, // nếu lưu JSON string thì để vậy
    story_point: { type: Number, default: 0 },
    attachments: { type: [String], default: [] },

    created_at: { type: Date, default: Date.now },
    completed_at: { type: Date, default: null },
    updated_at: { type: Date, default: Date.now },
    due_date_from: { type: Date, default: null },
    due_date_to: { type: Date, default: null },

    creator_id: { type: String, required: true },
    key: { type: String, required: true },
    team_id: { type: String, required: true }
  },
  {
    timestamps: { createdAt: 'created_at', updatedAt: 'updated_at' }
  }
)

const IssueModel = mongoose.model('Issue', IssueSchema)
export default IssueModel
